src/server.cpp

```diff
@@ -1394,20 +1464,147 @@ void CServer::CreateAndSendChatTextForAllConChannels ( const int      iCurChanID
+        // Send chat text to all connected clients ---------------------------------
+        for ( int i = 0; i < iMaxNumChannels; i++ )
+        {
+            if ( vecChannels[i].IsConnected() )
+            {
+                // send message
+                vecChannels[i].CreateChatTextMes ( strActualMessageText );
+            }
+        }
+    }
 
-    // Send chat text to all connected clients ---------------------------------
-    for ( int i = 0; i < iMaxNumChannels; i++ )
+}
+void CServer::InterpretAndExecuteChatCommand ( const int iCurChanID,
+                                               const QString& strChatText )
+{
+    QString strChatCmd = strChatText;
+    strChatCmd.remove ( 0, 3 );
+    QString ChanName = vecChannels[iCurChanID].GetName(); // to greet the user
+
+    if ( strChatCmd == strEduModePassword && bEduModeEnabled )
     {
-        if ( vecChannels[i].IsConnected() )
+        vecChannels[iCurChanID].SetAdmin( true );
+        vecChannels[iCurChanID].CreateChatTextMes ( "Hi " + ChanName.toHtmlEscaped() + " You are an admin now.");
+    }
+    else if ( strChatCmd == "ls" && vecChannels[iCurChanID].IsAdmin() )
+    {   // list status of channels
+        vecChannels[iCurChanID].CreateChatTextMes ( "Hi " + ChanName.toHtmlEscaped() + ". Getting all IDs and names...");
+        // generate var to send
+        QString blocked = "";
+        QString channelListText = "<br><b>Channels names and nums: </b><table><tr style=\"font-weight: bold;\"><td>ID</td><td>Display name</td><td>Blocked</td></tr>";
+        for ( int i = 0; i < iMaxNumChannels; i++ )
         {
-            // send message
-            vecChannels[i].CreateChatTextMes ( strActualMessageText );
+            if ( vecChannels[i].IsConnected() )
+            {
+                QString id = QString::number(i);
+                if (vecChannels[i].IsBlocked())
+                {
+                    blocked = "<b style='color:red;'>Yes</b>";
+                } else {
+                    blocked = "<span style='color:green;'>No</span>";
+                }
+
+                channelListText += "<tr><td><b>" + id.toHtmlEscaped() + "</b></td><td>" + vecChannels[i].GetName().toHtmlEscaped() + "</td><td>" + blocked + "</td></tr>";
+
+            }
+        }
+        channelListText += "</table>";
+        vecChannels[iCurChanID].CreateChatTextMes ( channelListText );
+    }
+    else if ( strChatCmd.startsWith ( "disableChat" ) && vecChannels[iCurChanID].IsAdmin() )
+    {
+        EduModeSetFeatureDisabled( EDU_MODE_FEATURE_CHAT, true );
+        vecChannels[iCurChanID].CreateChatTextMes ( "Disabled Chat." );
+    }
+    else if ( strChatCmd.startsWith ( "enableChat" ) && vecChannels[iCurChanID].IsAdmin() )
+    {
+        EduModeSetFeatureDisabled( EDU_MODE_FEATURE_CHAT, false );
+        vecChannels[iCurChanID].CreateChatTextMes ( "Enabled Chat." );
+    }
+    else if ( strChatCmd.startsWith ( "bl" ) && vecChannels[iCurChanID].IsAdmin() )
+    {   // block user
```

Comment:
Manually block a user --> do not mix him into the mix of others & do not let him hear the others.